gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaStreamingExtractor.java

```diff
@@ -139,6 +129,10 @@ public void shutdown() {
       log.error("Interrupted when attempting to shutdown metrics collection threads.");
     }
     this.shutdownRequested.set(true);
+    // In case hasOutstandingFlush, we need to manually nack the ackable to make sure the CountDownLatch not hang
```

Comment:
Can we move this block to the parent class inside its own shutdown() method and call super.shutdown() here?